src/pollution_model/run_medusa_2.py

```diff
@@ -625,16 +653,70 @@ def main(selected_polygon_gdf: gpd.GeoDataFrame,
     scenario_id = run_pollution_model_rain_event(engine, area_of_interest, rainfall_event)
     # Ensure pollution model data is being served by geoserver
     serve_pollution_model()
+
+    # Create new table recording users' history
+    create_table(engine, MedusaScenarios)
+    # Create the query object
+    query = MedusaScenarios(
+        scenario_id=scenario_id,
+        antecedent_dry_days=antecedent_dry_days,
+        average_rain_intensity=average_rain_intensity,
+        event_duration=event_duration,
+        rainfall_ph=rainfall_ph,
+        geometry=area_of_interest['geometry'].to_wkt().iloc[0]
+    )
+    # Execute the query
+    execute_query(engine, query)
+
     return scenario_id
 
 
+def retrieve_input_parameters(scenario_id: int) -> MedusaRainfallEventGeometry:
+    """
+    Retrieve input parameters for the current scenario id.
+
+    Parameters
+    ----------
+    scenario_id: int
+        The scenario ID of the pollution model run
+
+    Returns
+    -------
+    MedusaRainfallEvent: class
+        Rainfall event parameters for MEDUSA 2.0 model
+    """
+    # Connect to the database
+    engine = setup_environment.get_database()
+
+    # Read from database
+    medusa_scenarios = gpd.GeoDataFrame.from_postgis(
+        """SELECT * FROM medusa_scenarios""",
+        con=engine,
+        geom_col="geometry"
+    )
+
+    if scenario_id in medusa_scenarios['scenario_id'].unique():
+        return MedusaRainfallEventGeometry(
+            antecedent_dry_days=medusa_scenarios.loc[medusa_scenarios['scenario_id'] == scenario_id, 'antecedent_dry_days'].iloc[0],
+            average_rain_intensity=medusa_scenarios.loc[medusa_scenarios['scenario_id'] == scenario_id, 'average_rain_intensity'].iloc[0],
+            event_duration=medusa_scenarios.loc[medusa_scenarios['scenario_id'] == scenario_id, 'event_duration'].iloc[0],
+            rainfall_ph=medusa_scenarios.loc[medusa_scenarios['scenario_id'] == scenario_id, 'rainfall_ph'].iloc[0],
+            geometry=medusa_scenarios.loc[medusa_scenarios['scenario_id'] == scenario_id, 'geometry'].iloc[0]
+        )
+    else:
+        return None
```

Comment:
This is quite a complicated way because you download all of the data from the database. If you use a method similar to this snipper to BG_Flood https://github.com/GeospatialResearch/Digital-Twins/blob/7e9a0639827b8ddc19b16cbf83b3e352146f15d1/src/flood_model/bg_flood_model.py#L144-L179 where it uses a longer SQL query you can reduce the python workload.